ext/TrixiConvexECOSExt.jl

```diff
@@ -158,6 +201,93 @@ function Trixi.bisect_stability_polynomial(consistency_order, num_eig_vals,
         gamma_opt = [gamma_opt]
     end
 
+    undo_normalization!(gamma_opt, consistency_order, num_stage_evals)
+
+    return gamma_opt, dt
+end
+
+# Specialized routine for PERK4.
+# For details, see Section 4 in 
+# - D. Doehring, L. Christmann, M. Schlottke-Lakemper, G. J. Gassner and M. Torrilhon (2024).
+# Fourth-Order Paired-Explicit Runge-Kutta Methods
+# [DOI:10.48550/arXiv.2408.05470](https://doi.org/10.48550/arXiv.2408.05470)
+function Trixi.bisect_stability_polynomial_PERK4(num_eig_vals,
+                                                 num_stage_evals,
+                                                 dtmax, dteps, eig_vals, cS3;
+                                                 verbose = false)
+    dtmin = 0.0
+    dt = -1.0
+    abs_p = -1.0
+
+    # Construct stability polynomial for each eigenvalue
+    pnoms = ones(Complex{Float64}, num_eig_vals, 1)
+
+    # Init datastructure for monomial coefficients
+    gamma = Variable(num_stage_evals - 5)
```

Comment:
If there is a neat way to combine the optimization routines that you see we should go for it. Although then we should maybe provide $p$ and $E$ to the optimizer explicitly to avoid confusion.

Reply:
So I gave this a revisit and I think making this general is a little bit too confusing/makes the code not much shorter.